sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -1106,6 +1106,11 @@ public partial interface IMemberNameConverter
     {
         string? ConvertMemberName(System.Reflection.MemberInfo member);
     }
+    public partial interface IModelSerializable
+    {
+        object Deserialize(System.BinaryData data, Azure.Core.Serialization.ModelSerializerOptions options);
+        System.BinaryData Serialize(Azure.Core.Serialization.ModelSerializerOptions options);
```

Comment:
As we talked at the review: I am not sure about returning BinaryData from the interface. I think we should do it only from helpers, e.g. ModelSerializer.Serialize. The interface should allow writing many models to the same buffer and buffer pooling. I know that you said that we can do the former, but then it means the IModelSerializable interface is just for the top of the model graph, and I think it's might be a missed opportunity to ship a public interface (using/taking a good name), such that we can use it just at the top of the graph.

Reply:
This is being addressed in this PR https://github.com/Azure/azure-sdk-for-net/pull/37349